src/fem_solver/src/solver_db.hpp

```diff
@@ -36,32 +45,28 @@ class FsResultMatrix {
 };
 using FsMatrixVec = std::vector<FsResultMatrix>;
 
-class Coeficients {
- public:
-	Coeficients() = default;
- private:
+using ValType = float;
+using Matrix = boost::numeric::ublas::compressed_matrix<ValType>;
+
+class SolverHeatmap {
+public:
+	using Temperature = ValType;
+	SolverHeatmap() = default;
+	SolverHeatmap(const Matrix& matrix) {
+		FillData(matrix);
+	}
+	void FillData(const Matrix& matrix) {
+		assert(temperatures_.empty() && "Heatmap not adapted for rewriting");
+		for (unsigned i = 0; i < matrix.size1(); ++i)
+			temperatures_.push_back(matrix(i, 0));
+	}
+	void Print() const;
+private:
 	size_t max_index_ = 0;
```

Comment:
max index doesn't updates and not needed. Please provide getter only getter function